### DISCUSSION TITLE
Outside Temperature reporting when Mode is OFF

### DISCUSSION THREAD

**Discussion by MuzzaMcD** [revk/xESP32-Faikout#220] [Ideas]
I have recently swapped from the official Daikin controllers (BRP072C42) to the Faikin. 
Very gratefull for the work being done here.
One thing I notice is that the Faikin does not report Outside Temperature when the AC Mode is set to OFF.
I was previously doing some crude automation in HAssist in relation to Cooling the house respecting Solar output, Solar Forecast, Room temperature and Outside Temp, etc,
Right now its 42 outside and the only way I can see how hot the environment is where the Outside condenser is, is to switch that unit ON so that the Faikin updates the state of the thermistor there.
While I could have an independent outside temp sensor, I have a number of Daikin units for different rooms on different sides of the house and the temperature differentials through the day as the sun moves is substantial.
The old Daikin controller reports the outside temp in whole degrees Celsius at all times. This was very useful for me.
Is there a setting controlling this or can we include this in an update?
Regards, Murray

**Comment by revk** [comment]
I will have to check, I'm still on holiday for another day, but I don't recall anything specifically to not get temp when off.

Do you see it on the web interface? I.e. is this just an issue with reporting over HA?

**Reply by turbolooser** [reply]
Hmm... I do see the temps of my 3MXM40 outside unit also updated on 1 internal unit which is switched off (but 2 others are running). @MuzzaMcD Do you say that if all internal units are off this situation happens or does 1 out of 3 switched off is good enough for that? 

I have just checked my HA longterm stats and I do see continous updates to outside temps during all units being off.
Maybe this is model specific? Which outside unit model you have?

**Reply by MuzzaMcD** [reply]
FYI , extra info …
My setup here does not involve ‘multi-splits’ but a set of separate, single-split AC units we call back to back installs.
So inside units are FTXM25QVMA, each connected to an RXM25QVMA. All rooms are independent and not connected.
These units are rated at 2.5 kW, so think relatively small rooms. I have another  larger space  that has same setup but 7kW.
This is same ‘model’ but FTXM71QVMA / RXM71QVMA.
I have found the 3 physical internal thermistors in these units that equate to outside, liquid and room temperature sensors, and have ‘proved’ each with manual manipulation to see changes reported to Faikin and HAssist.
I can also fortunately easily swap the Daikin controller and the Faikin controller to compare functionality.
![fakin](https://github.com/revk/ESP32-Faikin/assets/72003592/574681d1-9227-4260-a018-e49813711bb4)


Have learnt a lot here from these discussions and am happy to feed back my experience if/when I can.

**Comment by MuzzaMcD** [comment]
> Do you see it on the web interface? I.e. is this just an issue with reporting over HA?

The three temperatures are showing on web interface (Home Liquid, Outside) And the Outside temp on web is not updating when Mode=Off.  Here is a picture ... :)

![OutsideTemp](https://github.com/revk/ESP32-Faikin/assets/72003592/2852c352-8cd6-44bb-96d1-076cff42fb4f)

**Comment by revk** [comment]
I can see we poll with `F9` to get status on `G9` and we definitely do that when power on.

You can set `dump` to see the messages. It is possible your unit does not report it when off?!

**Reply by MuzzaMcD** [reply]
OK I see that the outside temp is not updated via Faikin CMD F9 > G9 when MODE=OFF, So I swapped back to Daikin Controller (BRP072C42) and observed that outside temp **IS** updated when unit MODE=OFF ... however. ...  on closer inspection it is '**ONLY**' checked/updated exactly every hour!  So I guess the Daikin controller is sending some other type of command.    Need to see how I can find out how to see this.  
Any pointers on how to 'listen' to command response with Daikin Controller attached ?

**Reply by revk** [reply]
The rx line on a Faikin can be tapped on to a tx or rx line between aircon and controller, and `snoop` mode set and it will dump what it sees.

**Reply by MuzzaMcD** [reply]
A quick update on this.
I swapped back to using Daikin Controller (BRP072C42), just to do some more analysis.
I just observed the code sent every hour when my Daikin is Mode:Off. For my unit it was 'DH1000'.
While I don't have any more details on a return message, I believe (have observed) this command in my case is causing the unit to somehow 'wake from sleep' for a minute, or at least perform some sort of handshake with the outside condenser unit, ... which IN TURN is resulting in the outside Temperature to reflect an updated value (when this is checked next).
I'm not specifically asking for this to be implemented in Faikin code but just putting this here to add to the 'fountain of knowledge'!
I also suspect via **limited** observation, that the update of the outside temperature value  with this process is only in whole degrees (C). As opposed to observing this value updating by 0.5 degrees C when unit is actually operating. **(Not confirmed**)
I think a solution to this (for me anyway) is to write a simple automation to send the DH1000 code to the unit via MQTT when the unit is MODE:OFF when I want an updated outside temperature value. That might be every hour or 30 mins?

**Reply by revk** [reply]
Sounds like a good idea. If that works well, maybe I could add to Faikin code as standard.

**Comment by mac73dev** [comment]
@MuzzaMcD 
Great find!
What was the exact MQTT path/topic and command you were using for submitting the 'DH1000' code?

I am able to send some MQTT commands (mode=cool / mode=off) using the path "command/Faikin/mode/" to put my device on or off but I have no idea where to exactly inject this 'DH1000' request.

**Reply by MuzzaMcD** [reply]
If you go to the wiki page here:
https://github.com/revk/ESP32-Faikin/wiki
at the very bottom where it describes 'debugging' and reference point 4.
The process is **_command/nameofac/send_**    'code'
I would take extra care using this generally as blindly issuing 'raw' codes is probably not very ... hmmm 'efficient' ?
I spent quite a bit of time scanning and analysing TX and RX codes to find this particular one ... that is only issued as far as I know when the unit is MODE:OFF and only every hour.... and it does cause the unit to use a little more power for a few minutes b4 it goes back to 'sleep' ... could be more consequences apart from updating the outside temp value ???
 Anyway, hope you find it useful :)

**Reply by mac73dev** [reply]
Thanks. Just wanted to confirm that it also works with my unit.

**Reply by lochstar** [reply]
`DH1000` command also works for my units `FTXV35UVMA` and `FTXV60WVMA`. Looks like `0.5` degree results are being returned as well.

**Comment by Sonic-Amiga** [comment]
@MuzzaMcD Hello! Interesting controller BRP072; i only have BRP069s. Can you tell me what's the difference ? Is it "newstyle" cloud-only or supports "old" HTTP API ? Would be interesting to see /common/get_basic_info

**Reply by lochstar** [reply]
@Sonic-Amiga Mine supports the old HTTP API (haven't updated it).

See https://github.com/revk/ESP32-Faikin/discussions/544

**Reply by MuzzaMcD** [reply]
> Would be interesting to see /common/get_basic_info
@Sonic-Amiga 
```
curl --insecure -H "X-Daikin-uuid: xxxxxx" "https://192.168.1.29/common/basic_info" 

ret=OK,type=aircon,reg=th,dst=1,ver=1_16_0,rev=182179A,pow=0,err=0,location=0,name=%64%75%74,icon=0,method=home only,lpw_flag=0,adp_kind=3,pv=2,cpv=2,cpv_minor=00,led=0,en_setzone=1,mac=706655F8FB1C,ssid=DaikinAP79689,adp_mode=ap_run,en_hol=0,enlver=1.00,grp_name=,en_grp=0,en_secure=1,port=30050,id=,pw=% 
```
```
curl --insecure -H "X-Daikin-uuid: xxxxxx" "https://192.168.1.29/aircon/get_model_info"

ret=OK,model=0C86,type=N,pv=2,cpv=2,cpv_minor=00,mid=NA,humd=0,s_humd=0,acled=0,land=0,elec=0,temp=1,temp_rng=0,m_dtct=1,ac_dst=--,disp_dry=0,dmnd=0,en_scdltmr=1,en_frate=1,en_fdir=1,s_fdir=3,en_rtemp_a=0,en_spmode=0,en_ipw_sep=0,en_mompow=0,en_patrol=0,en_fdir2=0,en_filter_sign=0% 
```
Sorry, I'm not that well informed on differences between Daikin controllers. Except the APP and integrations were not to my liking and so I swapped to the one and only Faikin :)

**Reply by Sonic-Amiga** [reply]
That's okay, i just want to know what's out there because i am participating in RE effort.
So, your BRP072C42 is a Thai unit, is that correct (reg=th) ? I suggest 'th' is 'thailand'. BTW, what's the original app you're using ? Since it's Asia, i am assuming it's not Onecta (because Onecta is region-locked to EU). I also see interesting firmware version number of 1.16.0, while Onecta comes with 1,14,88. I say "comes with" because firmware is not downloaded from the Net; instead it's simply bundled in the app's APK,

**Reply by MuzzaMcD** [reply]
> So, your BRP072C42 is a Thai unit, is that correct (reg=th)

Correct. I'm in Australia and when I purchased a few BRP072C42 controllers locally years ago they were labelled Daikin Industries (Thailand) Ltd. 
At the time I downloaded an app called Daikin Mobile Controller which was tagged with  Developer: Daikin Industries (Thailand)
Later, after installing a Daikin Ceiling Cassette Unit in another room, I purchased a BRP15B61 controller for that. Expecting it would show up in the existing APP I was using ..... Of course not!  So I had to download a second APP. Daikin Airbase tagged with Developer: Daikin Australia. ?!?
I attempted some crude automation which was not very satisfactory, especially when the Firmware would mysteriously update by way of the Apps. The phone Apps were very 'pretty' but under the cover poor. Since these bad experiences and with other IOT devices I'm a BIG fan of local control.  Gave up with frustration and moved to Faikin to replace the BRP072C42 controllers.

**Reply by Sonic-Amiga** [reply]
@MuzzaMcD Hello! Are you still here i hope ? Can you instruct me on how to pull X-Daikin-uuid out of my app ? I managed to reflash my BRP069B41 and it became BRP072C42 .
Connection is https, i guess it's not subject to wireshark-ing

**Comment by Sonic-Amiga** [comment]
@MuzzaMcD Nevermind, i found it out. I actually don't need to pull it from the app; i found how this UUID is registered. HTTP interface doc updated. Thank you for your great info anyways. I managed to upgrade my BRP069B41, up and running, and recognizing new protocol 3.40

**Reply by lochstar** [reply]
Can you link me to the doc? I couldn't work this out the other day when I wanted to try mine.

**Reply by MuzzaMcD** [reply]
@lochstar if you mean how to register UUID?  then ..

`curl --insecure -H "X-Daikin-uuid: <uuid here>" -v "https://192.168.1.21/common/register_terminal?key= <13 digit key here>"`

13 digit key is from sticker on BRP072C42 controller
The -v is verbose response and is optional
Generate a UUID4 from say uuidgenerator.net and strip out hyphens.

**Comment by Sonic-Amiga** [comment]
The doc is here: https://github.com/revk/ESP32-Faikin/wiki/Daikin-online-controller-reverse-engineering

**Comment by icygame** [comment]
I too have observed the absence of outdoor temperature readings when mode=off - it broke a number of automations that previously used this datum. The same Daikin unit reported outdoor temperature even when off using the Daikin BRP072A42 (since swapped to another unit).

I see the discussion above of the manual polling approach for gleaning outdoor temp...is there any plan to add this to the faikin firmware?

**Reply by revk** [reply]
I'll have a look.

**Reply by icygame** [reply]
thx!

**Reply by icygame** [reply]
i've upgraded. won't be able to confirm behavior till it cools overnight!

**Reply by icygame** [reply]
confirming it seems to work perfectly. updating hourly when not operating.

**Reply by icygame** [reply]
still works for me....and no no spikes either.

**Comment by revk** [comment]
Ok I have issued a beta which should do `DH1000` every hour. Do test.

**Reply by lochstar** [reply]
Thanks, I'll try it out today. Should only need to poll when the unit it is powered off.

**Reply by MuzzaMcD** [reply]
> Do test.

Yep, looks good. Tested on two units here.  Only thing I noticed, and I think I saw this before, is that when a firmware update happens I see a spike in the reported outside temp in HAssist graph.  Looks like this:
![Screenshot 2024-12-06 at 16 22 06](https://github.com/user-attachments/assets/04191107-90b3-47a6-8da3-fca799831011)

**Reply by revk** [reply]
The spike is odd.

**Reply by lochstar** [reply]
I also get those spikes, but it's been occuring every night since I added the manual poll.

**Reply by lochstar** [reply]
The auto-polling seems to have stopped working at some point? Was it removed?

**Reply by revk** [reply]
I have not removed. On beta release, yes?

**Reply by lochstar** [reply]
Huh, looks like both units reverted to non-beta. Last update `2024-11-30`. I'll change them over and update. Thanks.

/edit
Yep, the polling has just started again. Interestingly the first poll was the "spike".

**Comment by revk** [comment]
Thanks
